### PR TITLE
refactor code to use class

### PR DESCRIPTION
**Pull request by yungez**
refactor code to use class.

### CODE REVIEW

File: src/baseRunner.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import * as utilties from "./utilities";
+import * as path from "path";
+
+export enum Option {
+    docker = "docker",
+    local = "local"
+}
+
+export class BaseRunner {
+    protected _outputChannel: vscode.OutputChannel;
```
Comment (zikalino) [inline src/baseRunner.ts]: maybe the whole class should be abstract?

File: src/baseRunner.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+                if (!utilties.validatePlaybook(playbook, this._outputChannel)) {
+                    return;
+                }
+
+                return this.runPlaybookInternal(playbook);
+            })
+    }
+
+    protected runPlaybookInternal(playbook) {
```
Comment (zikalino) [inline src/baseRunner.ts]: and this function could also be abstract

Comment (zikalino) [reply src/baseRunner.ts]: also playbook is a string, right? so we could add type

File: src/cloudShellRunner.ts [hunk 3]
```diff
@@ -11,81 +14,71 @@ import { setInterval, clearInterval } from 'timers';
 
 const tempFile = path.join(ost.tmpdir(), 'cloudshell' + vscode.env.sessionId + '.log');
 
-export function runPlaybook(outputChannel) {
-    const installedExtension: any[] = vscode.extensions.all;
+export class CloudShellRunner extends BaseRunner {
+
+    protected runPlaybookInternal(playbook) {
```
Comment (zikalino) [inline src/cloudShellRunner.ts]: add type and return value void

File: src/cloudShellRunner.ts [hunk 4]
```diff
@@ -11,81 +14,71 @@ import { setInterval, clearInterval } from 'timers';
...
+        const installedExtension: any[] = vscode.extensions.all;
 
-    let azureAccount: AzureAccount;
-    for (var i = 0; i < installedExtension.length; i++) {
-        const ext = installedExtension[i];
-        if (ext.id === Constants.AzureAccountExtensionId) {
-            azureAccount = ext.activate().then((azureAccount) => {
-                if (azureAccount) {
-                    startCloudShell(outputChannel);
+        let azureAccount: AzureAccount;
```
Comment (zikalino) [inline src/cloudShellRunner.ts]: maybe this could be a separate function

File: src/extension.ts [hunk 5]
```diff
@@ -2,28 +2,33 @@
...
-    let runcloudshell = vscode.commands.registerCommand('vscode-ansible.ansible-cloudshell', () => {
-        cloudshellRunner.runPlaybook(outputChannel);
-    });
+    context.subscriptions.push(vscode.commands.registerCommand('vscode-ansible.cloudshell', () => {
+        cloudShellRunner.runPlaybook();
+    }));
 
-    context.subscriptions.push(playbookinterminal);
-    context.subscriptions.push(runcloudshell);
+    context.subscriptions.push(vscode.window.onDidCloseTerminal((closedTerminal: vscode.Terminal) => {
```
Comment (zikalino) [inline src/extension.ts]: why is this needed here?
i thought it was enough to have it in TerminalExecutor, as i have tested it was called every time when terminal was closed.
is there any additional reason for this code?

Comment (yungez) [reply src/extension.ts]: terminalExecutor is changed, it is just a helper class. here call vscode.window.onDidCloseTerminal

File: src/terminalExecutor.ts [hunk 6]
```diff
@@ -2,72 +2,37 @@
 
 import * as vscode from 'vscode';
```
Comment (zikalino) [inline src/terminalExecutor.ts]: Maybe TerminalExecutor should become a part of TerminalRunner? it just has static functions which are used only from TerminalRunner

Comment (yungez) [reply src/terminalExecutor.ts]: it's separate as its functionality is quite independent. that's why separate it out
